Readme.md

```diff
@@ -42,6 +42,78 @@ $ npm install koa
  - [Introduction Screencast](http://knowthen.com/episode-3-koajs-quickstart-guide/) - An introduction to installing and getting started with Koa
 
 ## Example
+```js
+const Koa = require('koa');
+const app = new Koa();
+
+// logger
+
+app.use(function (ctx, next){
+  const start = new Date;
+  next(ctx).then(function() {
+    const ms = new Date - start;
+    console.log(`${ctx.method} ${ctx.url} - ${ms}`);
+  });
+});
+
+// response
+
+app.use(function (ctx){
+  ctx.body = 'Hello World';
+});
+
+app.listen(3000);
+```
+
+## Example with asyncFunction(babel-node required)
+
+```js
+const Koa = require('koa');
+const app = new Koa();
+
+// logger
+
+app.use(async (ctx, next) => {
+  const start = new Date;
+  await next(ctx);
+  const ms = new Date - start;
+  console.log(`${ctx.method} ${ctx.url} - ${ms}`);
+});
+
+// response
+
+app.use(function (ctx){
+  ctx.body = 'Hello World';
+});
+
+app.listen(3000);
+```
+
+## Example with generator
+
+```js
+const Koa = require('koa');
+const app = new Koa();
+
+// logger
+
+app.use(function *(ctx, next){
+  const start = new Date;
+  yield next(ctx);
+  const ms = new Date - start;
+  console.log(`${ctx.method} ${ctx.url} - ${ms}`);
+});
+
+// response
+
+app.use(function *(ctx){
+  ctx.body = 'Hello World';
+});
+
+app.listen(3000);
+```
+
+## Example with old version, Koa < 2
```

Comment:
I'd personally go with discouraging generator function from Koa 2.0. And start a migration process to eventually drop it altogether and remove `co` as a dependency. With Babel and promises (and sync'ed middlewares, there's really no need to support yet another async solution... which is hacky and was not meant for this situation. Generator functions not only lack debugging standard, but are totally misused (i.e. they confuse everyone as they were not meant as a control flow solution).